### PR TITLE
Requested Features added UI improved

### PR DESCRIPTION
**Pull request by fihlie**
In issue #382 we've decided to add some improvements on path page.

Following improvements have been made:
- Now waiting animation will be displayed on click to improve responsiveness.
- Finish Path Button is added, user can mark a path as finished.
- Effort keyword is changed to Difficulty to be more clear.
- Waiting animation will be displayed on rate path, rate difficulty and finish path buttons to improve responsiveness.

### CODE REVIEW

File: source/android/Renaissance/lib/http_services.dart [hunk 1]
```diff
@@ -181,6 +181,18 @@ class HttpService {
...
+  Future<bool> finish_path(String username, String id) async {
+    String url = baseUrl + '/path/finish-path/';
+    Response res = await post(Uri.parse(url),
+        headers: headers,
+        body:
+        jsonEncode({'username': username, 'path_id': id}));
+    if (res.statusCode == 200) {
+      return true;
+    }
+    return false;
```
Comment (kursatbakis) [inline source/android/Renaissance/lib/http_services.dart:193]: return response.statusCode == 200;

**Review by kursatbakis — CHANGES_REQUESTED** [review]
First class work. Better than ever! You've got it now.

File: source/android/Renaissance/lib/http_services.dart (hunk 1 above)
Comment (Hazerank) [reply source/android/Renaissance/lib/http_services.dart:193]: Wow, nice point Kürşat. Another efficient way of coding here!

**Review by Hazerank — APPROVED** [review]
The changes that you are done are important and works with the create path page update. So it is nice to put together a working application. Good Job.

File: source/android/Renaissance/lib/http_services.dart (hunk 1 above)
Comment (fihlie) [reply source/android/Renaissance/lib/http_services.dart]: Changes have been made, thanks for warning.

**Review by kursatbakis — APPROVED** [review]
thank you. we can merge it

**Pull request merged**